### DISCUSSION TITLE
Token replacements for linkDir

### DISCUSSION THREAD

**Discussion by kingofzeal** [cross-seed/cross-seed#709] [Ideas]
Loving how v6 works with regards to the `linkDir`. I'm wondering if it could be improved by supporting some replacement tokens as well - for example, the tracker name.

The primary reason is to make it easier to manage cross seeds with regards to data on the disk. My preferred setup would be to have all the cross seeds, each in their own directory, and configure the client to automatically remove them (with data) once a max seed time has expired - in my case using ruTorrent ratio groups, but this is largely independent of the actual client. If I then use hard links for each of those, the data will stay on the disk until the last torrent has been removed.

Currently, leveraging the `linkDir` allows me to separate the original torrent from cross seeds, but there's still the issue that the first cross seeded torrent will remove the files for all of them. If I could set a `linkDir` of something like "/downloads/cross-seed-data/{tracker}", then I believe it would work the way I outlined above as each torrent would have data in its own discrete folder, all hard linked to the same inode. The tracker name also appears to be something that could be on hand, since it's reported during a scan.

**Comment by ShanaryS** [comment]
Doesn't [`flatLinking: false`](https://www.cross-seed.org/docs/basics/options#flatlinking) solve this?

**Reply by kingofzeal** [reply]
You are... absolutely correct. I wasn't paying attention when I read that flag and set it to true, but now that I'm actually paying attention to the description, it's exactly what I'm describing for my use case.

I feel like there might be additional tokens that could be useful, but I admit I can't think of what they are.
